src/notification/notification.ts

```diff
@@ -17,8 +17,57 @@ import {sendTwitchMessage} from './twitch';
 import {updateRedis} from './redis';
 import {activateSmartthingsSwitch} from './smartthings';
 import {sendStreamLabsAlert} from './streamlabs';
+import {Print, logger} from '../logger';
 
-export function sendNotification(link: Link, store: Store) {
+type RequestFailed = {
+  failureReason: 'request_failed';
+  statusCode: number;
+};
+type Captcha = {failureReason: 'captcha'};
+type MaxPriceExceeded = {failureReason: 'max_price'; maxPrice: number};
+type OutOfStock = {failureReason: 'out_of_stock'};
+type BannedSeller = {failureReason: 'banned_seller'};
+type LinkPollError = {result: 'failure'} & (
+  | RequestFailed
+  | Captcha
+  | MaxPriceExceeded
+  | OutOfStock
+  | BannedSeller
+);
+type LinkPollSuccess = {result: 'in_stock'};
+
+export type LinkPollEvent = (LinkPollError | LinkPollSuccess) & {
+  link: Link;
+  store: Store;
+};
+export type SendNotification = (pollEvent: LinkPollEvent) => void;
+
+function assertNever(x: never): never {
+  throw new Error('Unexpected object: ' + x);
+}
+
+export function sendNotification(pollEvent: LinkPollEvent) {
+  const {link, store} = pollEvent;
+  if (pollEvent.result === 'failure') {
+    switch (pollEvent.failureReason) {
```

Comment:
```suggestion
  const {failureReason, link, result, store} = pollEvent;
  if (result === 'failure') {
    switch (failureReason) {
```

Might as well yank these out as well.